asset/src/kafka_sender/processor.ts

```diff
@@ -45,10 +45,10 @@ export default class KafkaSender extends BatchProcessor<KafkaSenderConfig> {
     }
 
     private getKey(msg: DataEntity): string|null {
-        if (!this.opConfig.id_field) return null;
+        const key = msg[this.opConfig.id_field] || msg.getMetadata('_key');
 
-        const key = msg[this.opConfig.id_field];
-        if (key && isString(key)) return key;
+        if (key === undefined) return null;
```

Comment:
Is this actually necessary? Wouldn't undefined achieve the same result?

Reply:
not sure, but the tests fail if undefined is returned.

Reply:
OK, just consider peter's feedback.